packages/connector/src/services/core/middleware/stream-address.ts

```diff
@@ -6,16 +6,8 @@ export function createStreamAddressMiddleware() {
     next: () => Promise<unknown>
   ): Promise<void> => {
     const { destination } = request.prepare
-    const accountId = streamServer.decodePaymentTag(destination)
-    if (!accountId) {
-      await next()
-      return
-    }
-
     // To preserve sender privacy, the accountId wasn't included in the original destination address.
-    const segments = destination.split('.')
-    segments.splice(-1, 0, accountId)
-    state.streamDestination = segments.join('.')
+    state.streamDestination = streamServer.decodePaymentTag(destination)
```

Comment:
Alot cleaner 👍 (still not 100% sure how this state should be shared between the middlewares, but looks good for now)